tests/trainer/test_trainer.py

```diff
@@ -889,6 +889,21 @@ def on_after_backward(self):
     assert not torch.isfinite(params).all()
 
 
+def test_detect_anomaly_nan(tmpdir):
+    class CurrentModel(BoringModel):
+        def forward(self, x):
+            x /= 0
+            return self.layer(x)
```

Comment:
we can't test it like this. the error for finite check with `terminate_on_nan` will trigger first.